pkg/migration/azure.go

```diff
@@ -28,10 +37,27 @@ func NewAzureMigratorFactory(cfg AzureMigrationConfig) (MigratorFactory, error)
 	}, nil
 }
 
-func (f *azureMigratorFactory) NewMigrator(clusterID string) (Migrator, error) {
+func (f *azureMigratorFactory) NewMigrator(cluster v1alpha3.Cluster) (Migrator, error) {
+	// Can't init the WC ctrl client here because I don't have the cluster object and so no control plane endpoint.
+
+	restConfig, err := f.config.TenantCluster.NewRestConfig(context.Background(), cluster.Name, cluster.Spec.ControlPlaneEndpoint.Host)
```

Comment:
I was wondering if we need to take `cluster.Spec.ControlPlaneEndpoint.Port` into account as well or if we can just have this as-is since we are only running this towards our own clusters?

Reply:
makese sense